### PR TITLE
Avoid creating illegal pointers

### PR DESCRIPTION
**Pull request by kroeckx**
Found by tis-interpreter

### CODE REVIEW

File: crypto/bn/bn_lib.c [hunk 1]
```diff
@@ -584,7 +584,8 @@ BIGNUM *BN_lebin2bn(const unsigned char *s, int len, BIGNUM *ret)
     ret->neg = 0;
     l = 0;
     while (n--) {
-        l = (l << 8L) | *(s--);
+        s--;
```
Comment (richsalz) [inline crypto/bn/bn_lib.c]: why not use ```*--s``` ?

**Comment by richsalz** [comment]
+1 no matter how you reply to my comment.

File: crypto/bn/bn_lib.c (hunk 1 above)
Comment (kroeckx) [inline crypto/bn/bn_lib.c]: Just so you can comment on it.

Comment (richsalz) [inline crypto/bn/bn_lib.c]: :)

**Comment by richsalz** [comment]
+1 still

**Pull request closed without merging**